### DISCUSSION TITLE
Support gRPC transport protocol for API

### DISCUSSION THREAD

**Discussion by MChorfa** [encoredev/encore#4] [Help] [answered]
Is there a way to generate Proto and/or GRPC resources for pre-defined set of operations?

**Comment by eandre** [comment, marked as answer]
Hey @MChorfa! It's not something we support right now but it's something we'd like to add support for.

**Reply by Rjvs** [reply]
Hoping you do this in an extensible way so that other options can also be selected (e.g. flatbuffers).

**Comment by dangdennis** [comment]
Looking at the way services are generated, are they RPC services underneath the hood?

**Reply by eandre** [reply]
Hey @dangdennis, can you elaborate a bit on what you mean? We don't currently use gRPC but generate regular HTTP handlers.

**Reply by dangdennis** [reply]
that was a dumb question after some digging, excuse me. initially, when looking at the way encore generates generate api routes, e.g. `ping.Hello`, it made me think you were using grpc underneath due to the naming convention. 

After perusing your docs and code, it's clear you're just just generating via static analysis tools

**Reply by eandre** [reply]
Not a dumb question, and glad it got answered by the docs! For public routes we now encourage specifying the route explicitly with the `path=/path/to/endpoint` syntax since it provides an API structure that's more familiar, but you can still leave it off if you're happy with the `/ping.Hello` style.
